pkg/disk/partition_table.go

```diff
@@ -1054,3 +1055,92 @@ func EnsureBootPartition(pt *PartitionTable, bootFsType FSType) error {
 	pt.Partitions = append(pt.Partitions, bootPart)
 	return nil
 }
+
+// AddPartitionsForBootMode creates partitions to satisfy the boot mode requirements:
+//   - BIOS/legacy: adds a 1 MiB BIOS boot partition.
+//   - UEFI: adds a 200 MiB EFI system partition.
+//   - Hybrid: adds both.
+//
+// The function will append the new partitions to the end of the existing
+// partition table therefore it is best to call this function early to put them
+// near the front (as is conventional).
+func AddPartitionsForBootMode(pt *PartitionTable, bootMode platform.BootMode) error {
+	switch bootMode {
+	case platform.BOOT_LEGACY:
+		// add BIOS boot partition
+		part, err := mkBIOSBoot(pt.Type)
+		if err != nil {
+			return err
+		}
+		pt.Partitions = append(pt.Partitions, part)
+		return nil
+	case platform.BOOT_UEFI:
+		// add ESP
+		part, err := mkESP(200*datasizes.MiB, pt.Type)
+		if err != nil {
+			return err
+		}
+		pt.Partitions = append(pt.Partitions, part)
+		return nil
+	case platform.BOOT_HYBRID:
+		// add both
+		bios, err := mkBIOSBoot(pt.Type)
+		if err != nil {
+			return err
+		}
+		esp, err := mkESP(200*datasizes.MiB, pt.Type)
+		if err != nil {
+			return err
+		}
+		pt.Partitions = append(pt.Partitions, bios, esp)
+		return nil
+	case platform.BOOT_NONE:
+		return nil
+	default:
+		return fmt.Errorf("invalid boot mode specified: %s", bootMode)
```

Comment:
(nitpick) maybe add a test for this too?

Reply:
Changed error message and added test.